src/clr-angular/forms/common/form.ts

```diff
@@ -28,4 +52,69 @@ export class ClrForm {
   markAsTouched() {
     this.markControlService.markAsTouched();
   }
+
+  @ContentChildren(ClrLabel, { descendants: true })
+  labels: QueryList<ClrLabel>;
+
+  @HostListener('submit')
+  onFormSubmit() {
+    this.markAsTouched();
+    if (isPlatformBrowser(this.platformId) && this.ariaLiveElement) {
+      this.invalidControls = Array.from(this.el.nativeElement.querySelectorAll('.ng-invalid'));
+      if (this.invalidControls.length > 0) {
+        this.invalidControls[0].focus();
+        this.updateAriaLive();
+      } else {
+        this.resetAriaLive();
+      }
+    }
+  }
+
+  ngOnInit() {
+    this.createAriaLiveElement();
+  }
+
+  private resetAriaLive() {
+    this.ariaLiveElement.innerHTML = '';
+  }
+
+  private updateAriaLive(): void {
+    if (this.invalidControls.length === 0) {
+      return;
+    }
+
+    this.resetAriaLive();
+
+    this.ariaLiveElement.textContent =
```

Comment:
Nit: we are using `.textContent` here to update the text content in this method, but in `resetAriaLive()` we are using `.innerHTML`. Can we stick to either `.innerHTML` or `.textContent` for consistency?

Reply:
Yes, I'll stick with `textContent`